pyvista/utilities/helpers.py

```diff
@@ -117,6 +117,11 @@ def convert_string_array(arr, name=None):
 
     """
     if isinstance(arr, np.ndarray):
+        # VTK default fonts only support ASCII. See https://gitlab.kitware.com/vtk/vtk/-/issues/16904
+        if np.issubdtype(arr.dtype, np.str_) and not ''.join(arr).isascii():  # avoids segfault
```

Comment:
The issue was that `convert_array()` dispatches to `convert_string_array()` for both unicode and bytes-like numpy arrays. Only the former can be joined with `''`, but the latter cannot contain non-ASCII characters by definition (I think).

There's still some weirdness: `convert_array()` converts object arrays to bytes-like numpy arrays. Our tests start from `np.arange(3).astype(object)` or something like that, which would contain ints rather than strings, but this also gets converted to strings. This is all unrelated to this fix, just wondering if everything there is as expected...